tests/test_hierarchy_inference.py

```diff
@@ -0,0 +1,149 @@
+import argparse
+import json
+from copy import deepcopy
+from collections import defaultdict
+
+from tqdm import tqdm
+import arrow
```

Comment:
Can we use the regular `time` library from stdlib? We're not using any special features of arrow